docs/case-studies/gitops.md

```diff
@@ -52,11 +123,167 @@ spec:
+
+Apply the `infra.yaml` to cluster, we can see that the MySQL in `infrastructure/` is automatically deployed:
+
+```shell
+kubectl apply -f clusters/infra.yaml
+```
+
+```shell
+$ vela ls
+
+APP   	COMPONENT       	TYPE      	TRAITS 	PHASE  	HEALTHY	STATUS	CREATED-TIME
+infra 	database-config 	kustomize 	       	running	healthy	      	2021-09-26 20:48:09 +0800 CST
+mysql 	mysql-controller	helm      	       	running	healthy	      	2021-09-26 20:48:11 +0800 CST
+└─  	mysql-cluster   	raw       	       	running	healthy	      	2021-09-26 20:48:11 +0800 CST
+```
+
+Apply the `apps.yaml` to cluster, we can see that the application in `apps/` is automatically deployed:
+
+```shell
+kubectl apply -f clusters/apps.yaml
+```
+
+```shell
+APP   	COMPONENT       	TYPE      	TRAITS 	PHASE  	HEALTHY	STATUS	CREATED-TIME
+apps  	apps            	kustomize 	       	running	healthy	      	2021-09-27 16:55:53 +0800 CST
+infra 	database-config 	kustomize 	       	running	healthy	      	2021-09-26 20:48:09 +0800 CST
+my-app	my-server       	webservice	ingress	running	healthy	      	2021-09-27 16:55:55 +0800 CST
+mysql 	mysql-controller	helm      	       	running	healthy	      	2021-09-26 20:48:11 +0800 CST
+└─  	mysql-cluster   	raw       	       	running	healthy	      	2021-09-26 20:48:11 +0800 CST
+```
+
+`curl` the Ingress of the app, we can see that the current version is `0.1.5` and the application is connected to the database successfully:
+
+```shell
+$ kubectl get ingress
+NAME        CLASS    HOSTS                 ADDRESS         PORTS   AGE
+my-server   <none>   testsvc.example.com   <ingress-ip>    80      162m
+
+$ curl -H "Host:testsvc.example.com" http://<ingress-ip>
+Version: 0.1.5
+
+$ curl -H "Host:testsvc.example.com" http://<ingress-ip>/db
+User: KubeVela
+Description: It's a test user
+```
+
+## Modify the config for GitOps trigger
+
+After the first deployment, we can modify the files in config repo to update the applications in the cluster.
+
+Modify the domain of the application's Ingress:
+
+```yaml
+...
+      traits:
+        - type: ingress
+          properties:
+            domain: kubevela.example.com
+            http:
+              /: 8089
+```
+
+Check the Ingress in cluster after a while:
+
+```shell
+NAME        CLASS    HOSTS                 ADDRESS         PORTS   AGE
+my-server   <none>   kubevela.example.com  <ingress-ip>    80      162m
+```
+
+The host of the Ingress has been updated successfully!
+
+## For developers
+
+![alt](../resources/dev-flow.jpg)
+
+For developers, they need to prepare an application code repository in addition to the KubeVela configuration repository. After the user updates the code in the application repository, a CI needs to be configured to automatically build the image and push it to the image registry. KubeVela listens for the latest images in the registry, automatically updates the image configuration in the repository, and finally updates the application configuration in the cluster.
```

Comment:
```suggestion
Developers writes the application source code and push it to a Git repo (aka app repo). Once app repo updates, the CI will build the image and push it to the image registry. KubeVela watches the image registry, and updates the image in config repo. Finally, it will apply the config to the cluster.
```